packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol

```diff
@@ -183,17 +185,29 @@ contract OVM_BridgeDepositBox is OVM_CrossDomainEnabled, OVM_Testable {
      * @param recipient L1 address that should receive the tokens.
      * @param l2Token L2 token to deposit.
      * @param amount How many L2 tokens should be deposited.
-     * @param maxFeePct Max fraction of the total `amount` that the depositor is willing to pay as a fee. This number is
-     * scaled by 1e18 to represent the percentage of the total transfer, deducted as a fee. Eg 5% would be 0.05e18.
+     * @param slowRelayFeePct Max fraction of `amount` that the depositor is willing to pay as a slow relay fee.
+     * @param instantRelayFeePct Fraction of `amount` that the depositor is willing to pay as a instant relay fee.
+     * @param quoteDeadline Timestamp, after which, this transaction will revert. The depositor will accept L1 LP fees
+     *      for any utilization amount up to this timestamp.
      */
     function deposit(
         address recipient,
         address l2Token,
         uint256 amount,
-        uint256 maxFeePct
+        uint64 slowRelayFeePct,
+        uint64 instantRelayFeePct,
+        uint64 quoteDeadline
     ) public onlyIfDepositsEnabled(l2Token) {
         require(isWhitelistToken(l2Token), "deposit token not whitelisted");
-        require(maxFeePct <= 1e18, "maxFeePct cannot be over 100% (represented as 1e18)");
+        require(
+            slowRelayFeePct + instantRelayFeePct <= 0.5e18,
+            "sum of slow & instant relayer fees can not exceed 50%"
+        );
+
+        // Note that the OVM's notion of `block.timestamp` is marginally different to the main ethereum L1 EVM. The OVM
+        // timestamp corresponds to the L1 timestamp of the last confirmed L1 ⇒ L2 transaction. If five minutes have
+        // passed since the last L1 ⇒ L2 transaction, the timestamp is automatically refreshed.
+        require(quoteDeadline >= getCurrentTime(), "deposit mined after deadline");
```

Comment:
Similar to above, deadline and the description is a little confusing. Is this deadline the quote timestamp? Or is it just some constraint on the quote timestamp, and the relayer uses the deposit timestamp for the quote?

Reply:
the deadline is a number up to which the depositor is willing to accept a quote. All we need to verify is that the current timestamp is less than or equal to the deadline such that the block inclusion time is before the deadline. The relayers will use the _block timestamp_ of the transaction to compute the LP fee for the deposit.

Reply:
I've updated the comments to be more clear.

Reply:
After discussing this we chose to change this to `quoteTimestamp`. PR has been updated accordingly